internal/contour/listener.go

```diff
@@ -194,6 +194,19 @@ func (lc *ListenerCache) httpsAddress() string {
 	return DEFAULT_HTTPS_LISTENER_ADDRESS
 }
 
+func (lc *listenerCache) HostHasTls(host string) bool {
```

Comment:
You cannot use values here without taking the lock. 

Please discuss  your design before sending code, this is the second time I have asked.